contracts/DecentralizedKV.sol

```diff
@@ -2,94 +2,125 @@
 pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
-import "./MerkleLib.sol";
-import "./BinaryRelated.sol";
+import "./libraries/MerkleLib.sol";
+import "./libraries/BinaryRelated.sol";
 
+/// @custom:upgradeable
+/// @title DecentralizedKV
+/// @notice The DecentralizedKV is a top base contract for the EthStorage contract. It provides the
+///         basic key-value store functionalities.
 contract DecentralizedKV is OwnableUpgradeable {
-    event Remove(uint256 indexed kvIdx, uint256 indexed kvEntryCount);
+    /// @notice Represents the metadata of the key-value .
+    /// @custom:field kvIdx  Internal address seeking.
+    /// @custom:field kvSize BLOB size.
+    /// @custom:field hash   Commitment.
+    struct PhyAddr {
+        uint40 kvIdx;
+        uint24 kvSize;
+        bytes24 hash;
+    }
 
+    /// @notice Enum representing different decoding types when getting key-value.
+    /// @custom:field RawData           Don't do any decoding.
+    /// @custom:field PaddingPer31Bytes Will remove the padding byte every 31 bytes.
     enum DecodeType {
         RawData,
         PaddingPer31Bytes
     }
 
-    uint256 public immutable storageCost; // Upfront storage cost (pre-dcf)
-    // Discounted cash flow factor in seconds
-    // E.g., 0.85 yearly discount in second = 0.9999999948465585 = 340282365167313208607671216367074279424 in Q128.128
-    uint256 public immutable dcfFactor;
-    uint256 public immutable startTime;
-    uint256 public immutable maxKvSize;
+    /// @notice Upfront storage cost (pre-dcf)
+    uint256 internal immutable STORAGE_COST;
+
+    /// @notice Discounted cash flow factor in seconds
+    ///         E.g., 0.85 yearly discount in second = 0.9999999948465585 = 340282365167313208607671216367074279424 in Q128.128
+    uint256 internal immutable DCF_FACTOR;
 
+    /// @notice The start time of the storage payment
+    uint256 internal immutable START_TIME;
+
+    /// @notice Maximum size of a single key-value pair
+    uint256 internal immutable MAX_KV_SIZE;
+
+    /// @custom:legacy
     /// @custom:spacer storageCost, dcfFactor, startTime, maxKvSize
     /// @notice Spacer for backwards compatibility.
-    uint256[4] public kvSpacers;
-
-    uint40 public kvEntryCount; // number of entries in the store
+    uint256[4] private kvSpacers;
```

Comment:
Is it necessary to apply the spacer naming rule?
> `The name of a spacer variable MUST be in the format spacer_<slot>_<offset>_<length> where <slot> is the original storage slot number, <offset> is the original offset position within the storage slot, and <length> is the original size of the variable.`

Reply:
Those spacers are temporary and will be deleted in the next version, so we are just using an array to represent them. After the contract is live on mainnet, we’ll follow the same practices as the code style.